stdlib/public/runtime/Paths.cpp

```diff
@@ -0,0 +1,554 @@
+//===--- Paths.cpp - Swift Runtime path utility functions -------*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+// Functions that obtain paths that might be useful within the runtime.
+//
+//===----------------------------------------------------------------------===//
+
+#include "swift/Runtime/Config.h"
+#include "swift/Runtime/EnvironmentVariables.h"
+#include "swift/Runtime/Debug.h"
+#include "swift/Runtime/Paths.h"
+#include "swift/Runtime/Win32.h"
+#include "swift/Threading/Once.h"
+
+#include <filesystem>
+
+#if !defined(_WIN32) || defined(__CYGWIN__)
+#include <sys/stat.h>
+
+#include <dlfcn.h>
+#else
+#define WIN32_LEAN_AND_MEAN
+#define NOMINMAX
+#include <windows.h>
+#include <psapi.h>
+#endif
+
+#include <cerrno>
+#include <cstdlib>
+#include <cstring>
+
+namespace {
+
+swift::once_t runtimePathToken;
+const char *runtimePath;
+
+swift::once_t rootPathToken;
+const char *rootPath;
+
+void _swift_initRuntimePath(void *);
+void _swift_initRootPath(void *);
+const char *_swift_getDefaultRootPath();
+char *_swift_getAuxExePathIn(const char *path, const char *name);
+const char *_swift_tryAuxExePath(const char *name, const char *path, ...);
+
+bool _swift_isPathSep(char ch) {
+#ifdef _WIN32
+  return ch == '/' || ch == '\\';
+#else
+  return ch == '/';
+#endif
+}
+
+bool _swift_exists(const char *path);
+
+#if !defined(_WIN32) || defined(__CYGWIN__)
+#define PATHSEP_STR "/"
+#define PATHSEP_CHR '/'
+#else
+#define PATHSEP_STR "\\"
+#define PATHSEP_CHR '\\'
+#endif
+
+}
+
+SWIFT_RUNTIME_EXPORT
+const char *
+swift_getRuntimePath()
+{
+  swift::once(runtimePathToken, _swift_initRuntimePath, nullptr);
+  return runtimePath;
+}
+
+SWIFT_RUNTIME_EXPORT
+const char *
+swift_getRootPath()
+{
+  swift::once(rootPathToken, _swift_initRootPath, nullptr);
+  return rootPath;
+}
+
+namespace {
+
+bool
+_swift_lookingAtLibSwift(const char *ptr, const char *base)
+{
+  // /some/path/to/some/thing/lib/swift/libswiftCore.dylib
+  //                         ^         ^
+  //                         |         +---- ptr
+  //                         +-------------- ptr - 10
+
+  return (ptr - base >= 10
+          && _swift_isPathSep(ptr[-10])
+          && std::strncmp(ptr - 9, "lib", 3) == 0
+          && _swift_isPathSep(ptr[-6])
+          && std::strncmp(ptr - 5, "swift", 5) == 0);
+}
+
+bool
+_swift_lookingAtBin(const char *ptr, const char *base)
```

Comment:
These helper functions are internal to this file, right? If so, they can be made `static` and don't need the `_swift_` prefix.

Reply:
It's inside an anonymous namespace already, so no need for `static`. It probably doesn't need the `_swift_` prefix, but that part won't hurt.